src/tokenizer.rs

```diff
@@ -418,8 +418,26 @@ impl<'a> Tokenizer<'a> {
                 quote_start if self.dialect.is_delimited_identifier_start(quote_start) => {
                     chars.next(); // consume the opening quote
                     let quote_end = Word::matching_end_quote(quote_start);
-                    let s = peeking_take_while(chars, |ch| ch != quote_end);
-                    if chars.next() == Some(quote_end) {
+                    let mut last_char = None;
```

Comment:
I think this code would be easier to understand if it were pulled out into its own function (`peeking_take_while_ident`?) which could be unit tested, to ensure it handles cases such as
```
foo "" bar "" 
foo bar""
```